src/jit/lclvars.cpp

```diff
@@ -236,6 +243,40 @@ void Compiler::lvaInitTypeRef()
     lvaInitArgs(&varDscInfo);
 
     //-------------------------------------------------------------------------
+    // Calculate the argument register usage.
+    //
+    // This will later be used for fastTailCall determination
+    //-------------------------------------------------------------------------
+
+    unsigned   floatingRegCount = 0;
+    unsigned   stackArgCount    = 0;
+    unsigned   stackSize        = 0;
+
+    auto incrementRegCount = [&floatingRegCount, &argRegCount](LclVarDsc* varDsc)
+    {
+        varDsc->IsFloatRegType() ? ++floatingRegCount : ++argRegCount;
+    };
+
+    for (LclVarDsc* varDsc, unsigned argNum = 0; argNum < info.compMethodInfo->args.numArgs; argNum++, varDsc++)
+    {
+        if (varDsc->lvRegNum != REG_STK)
+        {
+            incrementRegCount(varDsc);
+
+#ifdef (FEATURE_MULTIREG_ARGS && !WINDOWS_AMD64_ABI)
+            if (varDsc->lvOtherArgReg != REG_NA)
```

Comment:
although it's not clear it needs any platform `#ifdef`. If it does, please add a comment as to why.

---
In reply to: [124108646](https://github.com/jashook/coreclr/pull/6#discussion_r124108646) [](ancestors = 124108646,124107782)